### PR TITLE
Storing as ids rather than records (Countries, Activity, Categories)

### PR DESCRIPTION
**Pull request by mantagen**
- This PR allows url sharing, to populate components

### CODE REVIEW

File: client/src/CategorySelect.elm [hunk 1]
```diff
@@ -47,6 +64,23 @@ emptyCategory =
...
 
+toDict : List Category -> Dict.Dict String Category
+toDict categories =
+    categories
+        |> List.map (\category -> ( category.id, category ))
+        |> Dict.fromList
+
+
+getCategoryById : List Category -> CategoryId -> Category
+getCategoryById categories id =
```
Comment (Jbarget) [inline client/src/CategorySelect.elm]: dont fully understand why we convert to a Dict?

File: client/src/CountrySelect.elm [hunk 2]
```diff
@@ -16,14 +16,17 @@ type alias Index =
...
+
 type alias Model =
     { focused : Index
     , hovered : Index
     , menuOpen : Bool
-    , selected : Index
     , query : String
     , options : List Country
-    , selectedCountry : Maybe Country
+    , selectedCountry : Maybe CountryId
```
Comment (Jbarget) [inline client/src/CountrySelect.elm]: `selectedCountry` ---> `selected`

File: client/src/CountrySelect.elm [hunk 3]
```diff
@@ -134,20 +136,34 @@ view model =
...
+            Maybe.withDefault "" selectedCountry
 
         inputClass =
             classNames
                 [ ( "w-100 h2 pv2 pr3 br-pill ba b--solid b--blue", True )
                 , ( "pl3", not showInputFlag )
                 , ( "pl4", showInputFlag )
                 ]
+
+        inputValue =
```
Comment (Jbarget) [inline client/src/CountrySelect.elm]: nice 👍

File: client/src/Model/Model.elm [hunk 4]
```diff
@@ -38,11 +38,13 @@ type alias Model =
     , countries : List ( String, List String )
     , email : String
     , isLoggedIn : Bool
-    , activeCategory : Bool
     , countrySelect : CountrySelect.Model
     , activitySelect : ActivitySelect.Model
     , categorySelect : CategorySelect.Model
-    , activeCategory : CategorySelect.Category
+    , selectedCategories : List CategoryId
+    , selected : Maybe String
```
Comment (Jbarget) [inline client/src/Model/Model.elm]: what does this `selected` relate to?

File: client/src/ActivitySelect.elm [hunk 5]
```diff
@@ -197,20 +193,44 @@ view model { inputAlignment } =
...
             , activityMenu
                 model
                 menuClass
             ]
 
 
+getActivityName : Maybe ActivityId -> List Activity -> String
+getActivityName id activities =
+    case id of
+        Just a ->
```
[the remark this answers is not in the document]
Comment (mantagen) [reply client/src/ActivitySelect.elm]: `a` is not a type, `Just a` is creates variable `a`  if `id` isn't `Nothing`. And then you _know_ that `a` has a value. 
Have changed the wording to make it clearer :)

File: client/src/CategorySelect.elm (hunk 1 above)
Comment (mantagen) [reply client/src/CategorySelect.elm]: Just makes it a lot easier to deal with.
If we want to refactor at some point i'd say we should store all our `id`d data in Dicts and just use getters to get by id.

**Pull request merged**